sklearn/tree/tests/test_tree.py

```diff
@@ -1987,18 +1987,19 @@ def test_balance_property(criterion, Tree):
     assert np.sum(reg.predict(X)) == pytest.approx(np.sum(y))
 
 
-def test_poisson_zero_nodes():
+@pytest.mark.parametrize("seed", range(100))
+def test_poisson_zero_nodes(seed):
     # Test that sum(y)=0 and therefore y_pred=0 is forbidden on nodes.
     X = [[0, 0], [0, 1], [0, 2], [0, 3],
          [1, 0], [1, 2], [1, 2], [1, 3]]
     y = [0, 0, 0, 0, 1, 2, 3, 4]
     # Note that X[:, 0] == 0 is a 100% indicator for y == 0. The tree can
     # easily learn that:
-    reg = DecisionTreeRegressor(criterion="mse", random_state=1)
+    reg = DecisionTreeRegressor(criterion="mse", random_state=seed)
     reg.fit(X, y)
     assert np.amin(reg.predict(X)) == 0
     # whereas Poisson must predict strictly positive numbers
-    reg = DecisionTreeRegressor(criterion="poisson", random_state=1)
+    reg = DecisionTreeRegressor(criterion="poisson", random_state=seed)
     reg.fit(X, y)
     assert np.all(reg.predict(X) > 0)
```

Comment:
The domain of poisson includes 0:

```suggestion
    assert np.all(reg.predict(X) >= 0)
```

Reply:
No. Poisson has `y_true >= 0`, but `y_pred > 0`, otherwise the poisson deviance or log-likelihood is not defined.
With GLMs and the HGBT, this is easily enforced by the log-link `y_pred = exp(raw_prediction)`.

Reply:
`y_pred` is the expected value of the `y` variable assuming its aleatoric noise is modeled by a Poisson random variable. Even `y_train` can hold many exact zero values but `y_pred` should never be exactly zero.

Reply:
I am not sure if my above comment is very clear, helpful @thomasjpfan. I can try to rephrase if needed.

Reply:
I see how the math works out now. thank you! @ogrisel @lorentzenchr